### DISCUSSION TITLE
Need smoothing period in ADX

### DISCUSSION THREAD

**Discussion by myalgomate** [DaveSkender/Stock.Indicators#1015] [Ideas]
Hi Dave,

Referring to some of the charting platform they have 2 arguments and we need the same.

Also referring the reference link for the same.
https://www.investopedia.com/terms/a/adx.asp
https://finbold.com/guide/average-directional-index-definition/

![image](https://user-images.githubusercontent.com/76814842/216758606-79498419-91da-4c5f-b3f5-748ddb4519b2.png)

Regards,
Harsh

**Comment by DaveSkender** [comment]
Can you elaborate on "need to" here?  It sounds like you want to add a second parameter to separate out the DM smoothing period and the average DX period, which is possible, but why is it better (more options aren't always better)?  We're already smoothing properly in accordance with Welles Wilder's formula.

**Reply by myalgomate** [reply]
By using smoothing the results are totally different and we get more accurate data to know the trend.

This will be more important actually for ADX. I have seen some big algo working on the same logic.

![image](https://user-images.githubusercontent.com/76814842/216959092-0c2c9162-58eb-433f-a51a-ef8cbb0d13d6.png)

**Reply by DaveSkender** [reply]
Go ahead and add an Issue for tracking.  Not much harm to add an overload method here, if you’re seeing real outcomes improvements with it.

**Reply by DaveSkender** [reply]
Tracking in #1019
